### ISSUE TITLE
"Test" feature in cpbooster has some issue....

### ISSUE THREAD

**Issue by itsameaditya** [searleser97/cpbooster#9]
I am a Windows user and really liked the concept of cpbooster but here is one issue I am facing.....
So I parsed a problem from codeforces and all the test cases were downloaded properly .... I then the correct solution to the problem , but the cpbooster plugin says "Wrong Answer"

The answer cannot be wrong because it is an accepted and copy pasted code from codeforces hence there is some issue with the plugin . Here is a screenshot of the verdict:

![image](https://user-images.githubusercontent.com/73281787/118147474-3d4e4600-b42d-11eb-9b7f-f57dc7f06f96.png)

**Comment by itsameaditya** [comment]
Also just for confirmation I have manually compiled the code above and it gave the correct answer.

**Comment by searleser97** [comment]
Thanks for bringing this up!. I am aware of this issue but I haven't had the time to investigate how windows CMD works, apparently it does something weird with spaces and line-endings, is just weird. 

For now, I recommend you to use WSL (Windows Subsystem for Linux) it works perfectly with `cpbooster`.

**Comment by itsameaditya** [comment]
Can you tell me a way so that cpbooster does not check trailing or leading blank spaces and also to disable the TLE function .. It seems to have a few bugs..

**Comment by searleser97** [comment]
> Can you tell me a way so that cpbooster does not check trailing or leading blank spaces and also to disable the TLE function .. It seems to have a few bugs..

This bugs just appear when using windows CMD right ?

**Comment by itsameaditya** [comment]
I use the CMD to run "cpb clone" and then compile it using inbuilt neovim terminal

**Comment by searleser97** [comment]
Currently there is no way to disable the "leading or trailing whitespaces" check. For the TLE you can modify the default by adding the line `time-limit: 99999999`
this will tell `cpbooster` to consider a huge time limit or basically non-existent time-limit

**Comment by searleser97** [comment]
> I use the CMD to run "cpb clone" and then compile it using inbuilt neovim terminal

yeah, so, if you run neovim inside CMD then Neovim will use CMD as well for its built-in terminal. 

The inmediate solution is to use WSL in Windows instead of regular CMD. I suggest WSL with Ubuntu, since it will make it easier to install the packages.

**Comment by itsameaditya** [comment]
Can you please tell me how to do that.. I am not really good with OS stuff..

> > I use the CMD to run "cpb clone" and then compile it using inbuilt neovim terminal
> 
> yeah, so, if you run neovim inside CMD then Neovim will use CMD as well for its built-in terminal.
> 
> The inmediate solution is to use WSL in Windows instead of regular CMD. I suggest WSL with Ubuntu, since it will make it easier to install the packages.

**Comment by searleser97** [comment]
> Can you please tell me how to do that.. I am not really good with OS stuff..

There is a pretty nice tutorial from the oficial microsoft website here: https://docs.microsoft.com/en-us/windows/wsl/install-win10

**Comment by itsameaditya** [comment]
In cmd i just used to type "cpb clone", how do I do the same thing in the WSL thing?

**Comment by searleser97** [comment]
it works the same way in Linux (WSL) you just run `cpb clone` or `cpb (whatever command)`

However, since you are now in WSL (a different operative system inside Windows) you do have to install `cpbooster` again there.

You must install `NodeJs` and then do same as before, Just run `npm install cpbooster -g` and you will be ready to go.

**Comment by searleser97** [comment]
I recommend using `nvm` to install `NodeJs` https://github.com/nvm-sh/nvm#installing-and-updating

it is as easy as just running some simple commands;

1. `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash`
2. `nvm install node`
3. `npm install cpbooster -g`

**Comment by itsameaditya** [comment]
Does that mean i have to reinstall Neovim too right?

**Comment by searleser97** [comment]
> Does that mean i have to reinstall Neovim too right?

right, you can run this commands to get it installed:

- `sudo add-apt-repository ppa:neovim-ppa/stable`
- `sudo apt-get update`
- `sudo apt-get install neovim`

**Comment by itsameaditya** [comment]
its too complex dude... i have to rewrite my init.vim .... too much work to do...

**Comment by itsameaditya** [comment]
Also do you know any other plugin which is similar to cpbooster .. (for windows) i really loved this idea .....

**Comment by searleser97** [comment]
I recommend to backup your `init.vim` in a repository like github, so that you can just `git clone` it in any machine. In Linux you must locate this file in `/home/USERNAME/.config/nvim/init.vim`. For example: https://github.com/searleser97/nvim

It will only be an issue if you have some mappings or plugins that just work on Windows, but I doubt that you have something like that, most vim/neovim plugins were made thinking in linux first.

About other tools similar to `cpbooster`, I know some, but installation steps are more painful even in Linux, also not sure if any of them work for Windows, all of them have instructions for linux. I think it is because Windows requires to write more specific code to the platform, which makes it more complicated.

**Comment by searleser97** [comment]
Also, if you like to contribute fixing `cpbooster` for Windows, I'll appreciate it pretty much, otherwise, probably I will return to this in 3 or  4 months.

**Comment by itsameaditya** [comment]
> Also, if you like to contribute fixing `cpbooster` for Windows, I'll appreciate it pretty much, otherwise, probably I will return to this in 3 or 4 months.

I would love to but I am not as good as you.. but i will surely give it a shot

**Comment by itsameaditya** [comment]
> I recommend to backup your `init.vim` in a repository like github, so that you can just `git clone` it in any machine. In Linux you must locate this file in `/home/USERNAME/.config/nvim/init.vim`. For example: https://github.com/searleser97/nvim
> 
> It will only be an issue if you have some mappings or plugins that just work on Windows, but I doubt that you have something like that, most vim/neovim plugins were made thinking in linux first.
> 
> About other tools similar to `cpbooster`, I know some, but installation steps are more painful even in Linux, also not sure if any of them work for Windows, all of them have instructions for linux. I think it is because Windows requires to write more specific code to the platform, which makes it more complicated. Anyways, some of them are: `cf-tool`, `oj` (I don't remember more) (notice this tools do not have a plugin for vim/neovim. you could create it though)

What is oj? I cant find it anywhere..

**Comment by itsameaditya** [comment]
Just one more thing do i have to reintall c++ compiler as well for WSL? If so how do I do that?

**Comment by itsameaditya** [comment]
also I have reached till this step and cofigured neovim and all .. but the cpbooster "template" shows some error in WSL :
![image](https://user-images.githubusercontent.com/73281787/118476955-c74b2700-b72b-11eb-89c7-f953adbeca10.png)

**Comment by searleser97** [comment]
assumming you don't have any typo in your path, then the only issue I can think of is that probably you created the directory or file with some elevated permissions such as `root` permissions

**Comment by itsameaditya** [comment]
There is no typo for sure

> assumming you don't have any typo in your path, then the only issue I can think of is that probably you created the directory or file with some elevated permissions such as `root` permissions

There is no typo for sure.
Whats the fix for the 'root' permissions problem?

**Comment by itsameaditya** [comment]
Even WSL does not seem to show the correct results:

**Comment by searleser97** [comment]
> Even WSL does not seem to show the correct results:
> 
> ![image](https://user-images.githubusercontent.com/73281787/118517151-98957680-b754-11eb-80ff-a54dc27ef332.png)

This one is super weird, because I have tested `cpbooster` on WSL before 🤔. Can you open a new issue and describe the steps to reproduce this bug in detail ?

**Comment by itsameaditya** [comment]
> > Even WSL does not seem to show the correct results:
> > ![image](https://user-images.githubusercontent.com/73281787/118517151-98957680-b754-11eb-80ff-a54dc27ef332.png)
> 
> This one is super weird, because I have tested `cpbooster` on WSL before 🤔. Can you open a new **issue** and describe the steps to reproduce this bug in detail ?

You mean i shall describe how i reached this configuration  in setting up WSL and using cpbooster on it?

**Comment by searleser97** [comment]
Probably skip the WSL setup, just mention if you used "cpb clone" or "cpb create" or "cpb test <file> --add"

**Comment by searleser97** [comment]
Glad you got it working in WSL 🙂

 FYI, I  deleted the latest comments since they describe a different issue and were better summarized on issue: #11
